### PR TITLE
Cluster Restart support: graceful cluster state change to PASSIVE

### PR DESCRIPTION
**Pull request by mtopolnik**
In order to gracefully restart a Hazelcast cluster, it must first enter the PASSIVE state. Jet must prepare to enter this state by terminating all the running jobs.

This PR introduces the ability of Jet to prepare for the PASSIVE cluster state.

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/PrepareForPassiveClusterOperation.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+package com.hazelcast.jet.impl.operation;
+
+import com.hazelcast.jet.impl.JetService;
+import com.hazelcast.spi.Operation;
+
+/**
+ * Sent from the member that initiates a cluster state change to the master.
+ * When the operation completes, all jobs have been terminated.
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/PrepareForPassiveClusterOperation.java]: This comment is stale: the operation completes immediately.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/PrepareForPassiveClusterOperation.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+public class PrepareForPassiveClusterOperation extends Operation {
+
+    public PrepareForPassiveClusterOperation() {
+    }
+
+    @Override
+    public void run() {
+        if (!getNodeEngine().getClusterService().isMaster()) {
+            return;
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/PrepareForPassiveClusterOperation.java]: We should probably throw here: if the operation is sent to incorrect member, it should be retried to correct master.

**Comment by mtopolnik** [comment]
verify

**Comment by mtopolnik** [comment]
verify

**Comment by mtopolnik** [comment]
verify

**Pull request merged**